File: Python/Product of Array Except Self.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        result = []
        n = len(nums)
        left = 1
        for i in range(n):
            result.append(left)
            left = left*nums[i]
        right = 1
        for i in range(n-1,-1,-1):
            result[i] *= right
            right *= nums[i]
        return result
       

File: php/Product of Array Except Self.php
<?php
class Solution {
function productExceptSelf($nums) {
    $result = [];
    $left = 1;
    $n = count($nums);
    for ($i = 0; $i < $n; $i++){
        array_push($result,$left);
        $left = $left*$nums[$i];
    }
    $right = 1;
    for ($i = $n-1; $i >= 0;$i--){
        $result[$i] *= $right;
        $right *= $nums[$i];
    }
    return $result;
}
}